### PR TITLE
Fix undefined value checks in use_http_client_2

### PR DESCRIPTION
**Pull request by alexmaras**
as mentioned, 2 of the bugs I hit trying to run the new UI build on my home instance.

### CODE REVIEW

**Review by SleeplessOne1917 — APPROVED** [review]
Besides one small nitpick, LGTM.

File: src/shared/components/post/post-listings.tsx [hunk 1]
```diff
@@ -115,7 +115,7 @@ export class PostListings extends Component<PostListingsProps, any> {
 
   removeDuplicates(): PostView[] {
     // Must use a spread to clone the props, because splice will fail below otherwise.
-    const posts = [...this.props.posts];
+    const posts = [...this.props.posts].filter((empty) => empty);
```
Comment (dessalines) [inline src/shared/components/post/post-listings.tsx:118]: https://stackoverflow.com/questions/281264/remove-empty-elements-from-an-array-in-javascript

Seems like `.filter(Boolean)` is a bit simpler.

**Comment by dessalines** [comment]
For some reason husky didn't get set up properly on your install, which means it wasn't able to pretty up the files for you. You'll have to run `yarn prettier -w ...` on the files to get them to pass lint.

**Comment by alexmaras** [comment]
Thanks @dessalines & @SleeplessOne1917  - sorted out Husky, and linted. Also swapped to null coalescing.

**Review by dessalines — APPROVED** [review]
Looks good, thanks!

**Pull request merged**